src/Sentry/Protocol/Replay.cs

```diff
@@ -0,0 +1,88 @@
+using Sentry.Extensibility;
+using Sentry.Internal;
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Protocol;
+
+/// <summary>
+/// Sentry Replay context interface.
+/// </summary>
+/// <example>
+/// {
+///     "contexts": {
+///         "replay": {
+///             "replay_id": "12312012123120121231201212312012"
+///         }
+///     }
+/// }
+/// </example>
+/// <see href="https://develop.sentry.dev/sdk/data-model/event-payloads/contexts/#replay-context"/>
+public sealed class Replay : ISentryJsonSerializable, ICloneable<Replay>, IUpdatable<Replay>
+{
+    /// <summary>
+    /// Tells Sentry which type of context this is.
+    /// </summary>
+    public const string Type = "replay";
+
+    /// <summary>
+    /// The name of the runtime.
+    /// </summary>
+    public SentryId? ReplayId { get; set; }
+
+    /// <summary>
+    /// Clones this instance.
+    /// </summary>
+    public Replay Clone()
+    {
+        var response = new Replay();
+
+        response.UpdateFrom(this);
+
+        return response;
+    }
+
+    /// <summary>
+    /// Updates this instance with data from the properties in the <paramref name="source"/>,
+    /// unless there is already a value in the existing property.
+    /// </summary>
+    public void UpdateFrom(Replay source)
+    {
+        ReplayId ??= source.ReplayId;
+    }
+
+    /// <summary>
+    /// Updates this instance with data from the properties in the <paramref name="source"/>,
+    /// unless there is already a value in the existing property.
+    /// </summary>
+    public void UpdateFrom(object source)
+    {
+        if (source is Replay response)
+        {
+            UpdateFrom(response);
+        }
+    }
+
+    /// <inheritdoc />
+    public void WriteTo(Utf8JsonWriter writer, IDiagnosticLogger? logger)
+    {
+        writer.WriteStartObject();
+
+        writer.WriteString("type", Type);
+        writer.WriteSerializableIfNotNull("replay_id", ReplayId, logger);
```

Comment:
question: Could it be problematic if `ReplayId` does not have a value?

`replay_id` is _required_.
But we only write it if it has a value.

Could we run into this issue, or this this guarded against somewhere else?

Reply:
It doesn't appear to cause any trouble. In fact, Relay doesn't seem to care what is in the Replay context - it reads the value from the DSC and populates the Replay context automatically based on what it finds there. 

I see what you're saying though. According to [the docs](https://develop.sentry.dev/sdk/data-model/event-payloads/contexts/#replay-context) it's required. The docs also say:
> This should not be sent by SDKs in most cases, as the Dynamic Sampling Context will be plucked in relay, but for current versions of python and PHP these SDKs will send this context as they do not currently support the envelope endpoint.

So probably we should remove the Replay context altogether.